tests/admin_views/test_autocomplete_view.py

```diff
@@ -0,0 +1,137 @@
+from __future__ import unicode_literals
+
+import json
+
+from django.contrib import admin
+from django.contrib.admin import site
+from django.contrib.admin.tests import AdminSeleniumTestCase
+from django.contrib.admin.views.autocomplete import AutocompleteJsonView
+from django.contrib.auth.models import Permission, User
+from django.http import Http404
+from django.test import RequestFactory, override_settings
+from django.urls import Resolver404, resolve, reverse, reverse_lazy
+from django.utils.six import text_type
+
+from .admin import AnswerAdmin, QuestionAdmin
+from .models import Answer, Author, Authorship, Book, Question
+from .tests import AdminViewBasicTestCase
+
+
+class AuthorAdmin(admin.ModelAdmin):
+    search_fields = ['id']
+
+
+class AuthorshipInline(admin.TabularInline):
+    model = Authorship
+    autocomplete_fields = ['author']
+
+
+class BookAdmin(admin.ModelAdmin):
+    inlines = [AuthorshipInline]
+
+
+site.register(Question, QuestionAdmin)
+site.register(Answer, AnswerAdmin)
+site.register(Author, AuthorAdmin)
+site.register(Book, BookAdmin)
+
+
+class AutocompleteJsonViewTest(AdminViewBasicTestCase):
+    as_view_args = {'model_admin': AnswerAdmin(Question, site)}
+
+    url = reverse_lazy('admin:admin_views_question_autocomplete')
+
+    @classmethod
+    def setUpTestData(cls):
+        cls.user = User.objects.create_user(username='user', password='secret',
```

Comment:
Use hanging indent as described in [Python coding style](https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/coding-style/#python-style).